src/librustc_driver/lib.rs

```diff
@@ -151,6 +157,44 @@ pub fn run<F>(run_compiler: F) -> isize
     0
 }
 
+#[cfg(not(feature="llvm"))]
+pub struct NoLLvmMetadataLoader;
+
+#[cfg(not(feature="llvm"))]
+type LlvmMetadataLoader = NoLLvmMetadataLoader;
+#[cfg(feature="llvm")]
+type LLvmMetadataLoader = rustc_trans::LlvmMetadataLoader;
+
+#[cfg(not(feature="llvm"))]
+impl MetadataLoader for NoLLvmMetadataLoader {
+    fn get_rlib_metadata(&self, _: &Target, filename: &Path) -> Result<ErasedBoxRef<[u8]>, String> {
```

Comment:
Was this copied wholesale from `rustc_trans`? Does that mean `ar::Archive` is independent of LLVM?
That could make `rustc_metadata` target-agnostic in the non-dynamic-library case!
cc @alexcrichton